server/models/helpers/NotificationHelper.ts

```diff
@@ -146,64 +153,75 @@ export default class NotificationHelper {
     notificationType: NotificationEventType;
     onlySubscribers: boolean;
     actorId: string;
-  }): Promise<User[]> => {
-    // First find all the users that have notifications enabled for this event
-    // type at all and aren't the one that performed the action.
-    let recipients = await User.findAll({
```

Comment:
I also changed this to avoid having to query all of the users in a workspace when `onlySubscribers: true` (the most common case)